Enable Stress mode environment variable

tarekgh opened the pull request:
To write any stress test, you may mark the stress test case with one of the attributes:

```
[ConditionalFact(typeof(TestEnvironment), nameof(TestEnvironment.IsStressModeEnabled))]
[ConditionalTheory(typeof(TestEnvironment), nameof(TestEnvironment.IsStressModeEnabled))]
```

and set the value of the environment variable COREFX_STRESS to 1 or true

tarekgh commented:
CC @weshaggard @stephentoub @joshfree

stephentoub commented on src/System.Net.Http/tests/FunctionalTests/HttpClientMiniStressTest.cs:
```diff
@@ -16,9 +16,8 @@ namespace System.Net.Http.Functional.Tests
-        private static bool HttpStressEnabled => Configuration.Http.StressEnabled;
-
-        [ConditionalTheory(nameof(HttpStressEnabled))]
+        // IsStressModeEnabled is true when the environment variable COREFX_STRESS is set to 1 or true
```
I don't think these comments are necessary, but otherwise looks good.  You might see if adding this comment instead as an XML comment on IsStressModeEnabled (and ensuring that project builds an XML comment file) produces a better experience.

tarekgh commented:
@stephentoub I have removed the comments and updated the XML doc tags on the IsStressModeEnabled but I couldn't generate the Xml doc file from CoreFx.Private.TestUtilities because we'll needed to Xml document all public APIs there.

stephentoub commented:
Thanks!
